mypy/nodes.py

```diff
@@ -2668,6 +2668,43 @@ def deserialize(cls, data: JsonDict) -> 'TypeAlias':
                    no_args=no_args, normalized=normalized)
 
 
+class PlaceholderTypeInfo(SymbolNode):
+    """Temporary node that will later become a type but is incomplete.
+
+    These are only present during semantic analysis when using the new
+    semantic analyzer. These are created if some dependencies of a type
+    definition are not yet complete. Their purpose is to allow
+    PlaceholderType instances to be created for types that refer to
+    incomplete types. Example where this is useful:
+
+      class C(Sequence[C]): ...
+
+    We create a PlaceholderTypeInfo for C so that the type C in
+    Sequence[C] can be bound. Without a placeholder, the class
+    definition couldn't be analyzed.
```

Comment:
After some thinking just this use case is not enough motivation for introducing `PlaceholderType` and `PlaceholderTypeInfo`. Since we just need the top-level of every base to have a valid `TypeInfo`, we can just reuse the `ForwardRef` and then resolve it on the subsequent iterations using local lookup of the name `'C'`. The real motivation at this step IMO is not passing local lookup function to `typeanal.py` (and of course recursive named tuples, aliases etc. in future). I agree this is a step in the right direction, but I think this should be explicitly mentioned somewhere in the docstring (if you agree).